File: Two Pointer/Two Sum II/optimal.java
class Solution {
    public int[] twoSum(int[] num, int target) {
        int left = 0;
        int right = num.length - 1;
        int res[] = {-1,-1};
        
        while(left < right){
            int currSum = num[left]+num[right];
            if(currSum == target){
                res[0] = left+1;
                res[1] = right+1;
                return res;
            }else if(currSum > target){
                right--;
            }else{
                left++;
            }
        }
        return res;
    }
}

File: Hash Map/Two sum/optimal.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        
        for i,val in enumerate(nums):
            
            tg = target - val
            
            if tg in map:
                
                return [map[tg],i]
            
            map[val] = i
            
        